### PR TITLE
docs: rework docs around errors

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

> Require https://github.com/withastro/docs/pull/9157

People keep removing errors, we cannot change people, but we can make it acceptable to remove errors if that's what more intuitive.

I also took this opportunity to refine some parts of the document

## Testing

N/A

## Docs

It's all docs

### CODE REVIEW

File: packages/astro/src/core/errors/README.md [hunk 1]
```diff
@@ -89,6 +90,19 @@ Example:
 
 The `@message` property is intended to provide slightly more context when it is helpful: a more descriptive error message or a collection of common messages if there are multiple possible error messages. Try to avoid making substantial changes to existing messages so that they are easy to find for users who copy and search the exact content of an error message.
 
+### Removing errors
+
+If an error is no longer relevant, it can deprecated by adding a `@deprecated` tag to the JSDoc comment with a message that will be shown in the docs. This is useful for users on previous versions who might still encounter the error to know that perhaps upgrading to a newer version of Astro would solve their issue.
```
Comment (bluwy) [inline packages/astro/src/core/errors/README.md:95]: IIUC, an error is no longer relevant and marked deprecated if it's still being used by the codebase, but in a deprecated logic path?

If we no longer use the error, would we completely remove it?

File: packages/astro/src/core/errors/README.md [hunk 2]
```diff
@@ -21,8 +21,8 @@ Message:
 
 - Begin with **what happened** and **why**. (ex: `Could not use {feature} because Server-side Rendering is not enabled.`)
 - Then, **describe the action the user should take**. (ex: `Update your Astro config with `output: 'server'` to enable Server-side Rendering.`)
-- Although this does not need to be as brief as the `title`, try to keep sentences short, clear and direct to give the reader all the necessary information quickly as possible.
-- Instead of writing a longer message, consider using a `hint`.
+- Although this does not need to be as brief as the `title`, try to keep sentences short, clear and direct to give the reader all the necessary information quickly as possible. Users should be able to skim the message and understand the problem and solution.
+- If your message is too long, or the solution is not guarantee to work, use the `hint` property to provide more information.
```
Comment (delucis) [inline packages/astro/src/core/errors/README.md:25]: Typo:
```suggestion
- If your message is too long, or the solution is not guaranteed to work, use the `hint` property to provide more information.
```

File: packages/astro/src/core/errors/README.md [hunk 3]
```diff
@@ -44,10 +44,10 @@ If you are unsure about anything, ask [Erika](https://github.com/Princesseuh)!
 
 ### Shape
 
-- **Names are permanent**, and should never be changed, nor deleted. Users should always be able to find an error by searching, and this ensures a matching result. When an error is no longer relevant, it should be deprecated, not removed.
+- **Names are permanent**, and should never be changed. Users should always be able to find an error by searching, and this ensures a matching result.
 - Contextual information may be used to enhance the message or the hint. However, the code that caused the error or the position of the error should not be included in the message as they will already be shown as part of the error.
 - Do not prefix `title`, `message` and `hint` with descriptive words such as "Error:" or "Hint:" as it may lead to duplicated labels in the UI / CLI.
-- Dynamic error messages must use the following shape:
+- Dynamic error messages **must** use the following shape:
```
Comment (delucis) [inline packages/astro/src/core/errors/README.md:50]: 😄

**Comment by ematipico** [comment]
How would docs rework this part?

File: packages/astro/src/core/errors/README.md (hunk 1 above)
Comment (bluwy) [reply packages/astro/src/core/errors/README.md:95]: I see! I think I'm confused in deciding when would a deprecation message be required or not required then. For me if I removed a feature that no longer uses an error, I could remove the error directly. And if I tweak how a feature works that the error is no longer relevant, I'd use a deprecation message?

I think some examples would be nice, but not a big deal if not. Was worried if someone could misjudge when to do which.

**Comment by Princesseuh** [comment]
> How would docs rework this part?

I'm not sure I understand what you mean?

File: packages/astro/src/core/errors/README.md (hunk 3 above)
Comment (Princesseuh) [reply packages/astro/src/core/errors/README.md:50]: next step will be in all caps

**Comment by ematipico** [comment]
When we remove an error that was previously published, are we going to send the users to an URL that is 404?

**Comment by Princesseuh** [comment]
> When we remove an error that was previously published, are we going to send the users to an URL that is 404?

No, see the linked docs PR

**Comment by ematipico** [comment]
Thank you!

**Review by ematipico — APPROVED** [review]

**Review by delucis — APPROVED** [review]

**Review by bluwy — APPROVED** [review]

**Pull request merged**